signing/vrfsigner.go

```diff
@@ -1,21 +1,25 @@
 package signing
 
 import (
+	"encoding/binary"
+
 	"github.com/oasisprotocol/curve25519-voi/primitives/ed25519"
 	"github.com/oasisprotocol/curve25519-voi/primitives/ed25519/extra/ecvrf"
 )
 
-var _ Signer = VRFSigner{}
-
 // VRFSigner is a signer for VRF purposes.
 type VRFSigner struct {
 	privateKey []byte
 	pub        *PublicKey
+
+	nonce uint64
 }
 
 // Sign signs a message for VRF purposes.
 func (s VRFSigner) Sign(msg []byte) []byte {
-	return ecvrf.Prove(ed25519.PrivateKey(s.privateKey), msg)
+	nonceMsg := make([]byte, 8)
+	binary.LittleEndian.PutUint64(nonceMsg, s.nonce)
```

Comment:
i think everything should be encoded with scale

Reply:
I cannot pass in anything besides `[]bytes` because the `common/types` package depends on `signing` so using any concrete type instead of `[]bytes` will introduce a cyclic dependency. The proper fix imo for this would be to not have `common/types` depend on `signing` but have it the other way round, but this requires a lot of changes.

I could pass in `scale.Encodable`, encode it with scale and then sign it, but we don't use scale in beacon weakcoin. Proposals are e.g. encoded like this before signing them: https://github.com/spacemeshos/go-spacemesh/blob/vrf-nonce-invokations/beacon/weakcoin/weak_coin.go#L341-L358

I feel like changing all of that to `scale` encoding is out of scope for this PR. What would do you say?

Reply:
i think this is irrelevant if more isolated approach will be used, lets get back to this if there will be disagreement about that